### PR TITLE
[SPARK-16403][Examples] Cleanup to remove unused imports, consistent style, minor fixes

### PR DESCRIPTION
**Pull request by BryanCutler**
## What changes were proposed in this pull request?

Cleanup of examples, mostly from PySpark-ML to fix minor issues:  unused imports, style consistency, pipeline_example is a duplicate, use future print funciton, and a spelling error.  
- The "Pipeline Example" is duplicated by "Simple Text Classification Pipeline" in Scala, Python, and Java.  
- "Estimator Transformer Param Example" is duplicated by "Simple Params Example" in Scala, Python and Java
- Synced random_forest_classifier_example.py with Scala by adding IndexToString label converted
- Synced train_validation_split.py (in Scala ModelSelectionViaTrainValidationExample) by adjusting data split, adding grid for intercept.
- RegexTokenizer was doing nothing in tokenizer_example.py and JavaTokenizerExample.java, synced with Scala version
## How was this patch tested?

local tests and run modified examples

### CODE REVIEW

**Comment by BryanCutler** [comment]
cc @MLnick @yanboliang - either of you mind taking a look?  Thanks!

File: examples/src/main/java/org/apache/spark/examples/ml/JavaPipelineExample.java [hunk 1]
```diff
@@ -1,88 +0,0 @@
...
-import org.apache.spark.ml.feature.Tokenizer;
-import org.apache.spark.sql.Dataset;
-import org.apache.spark.sql.Row;
-// $example off$
-import org.apache.spark.sql.SparkSession;
-
-/**
- * Java example for simple text document 'Pipeline'.
- */
-public class JavaPipelineExample {
```
Comment (srowen) [inline examples/src/main/java/org/apache/spark/examples/ml/JavaPipelineExample.java]: I believe you but for reference what are these deleted files each duplicating?

Comment (BryanCutler) [inline examples/src/main/java/org/apache/spark/examples/ml/JavaPipelineExample.java]: Yeah, I'll update that in the description too.  Each "Pipeline Example"  is duplicated by "Simple Text Classification Pipeline" in Scala, Python and Java.  The test data is slightly different, but "Pipeline Example" gives false negatives while "Simple Text.." does not.

**Comment by srowen** [comment]
I'm going to merge this to master only if nobody has further comments.

**Comment by BryanCutler** [comment]
Thanks for the review Sean!

**Comment by BryanCutler** [comment]
I realized that the "PipelineExample"s are included in the docs, while the "SimpleTextClassificationExample"s are not, so it might be better to keep those instead.  I just changed the data and regularization value to that of "SimpleTextClassificationExample" which gives correct predictions (it looks like these examples were updated at one time by DB to fix this, but the change was not put into the doc example).

**Comment by BryanCutler** [comment]
I noticed that in a similar vein, SimpleParamsExample was a duplicate of EstimatorTransformerParamExample in Scala, Java, and Python, and removed those.

**Comment by srowen** [comment]
Merged to master

**Comment by BryanCutler** [comment]
Thanks @srowen!

**Pull request closed without merging**